File: algorithm/typescript/src/valid_anagram.test.ts
import { valid_anagram } from "./valid_anagram";

test("test valid_anagram", () => {
  let testCases: [string, string, boolean][] = [
    ["anagram", "nagaram", true],
    ["rat", "car", false],
    ["listen", "silent", true],
    ["hello", "world", false],
  ];

  testCases.forEach(([s, t, expected]) => {
    expect(valid_anagram(s, t)).toEqual(expected);
  });
});

File: algorithm/python/src/valid_anagram_test.py
# Test cases for isAnagram
from src.valid_anagram import isAnagram, isAnagram2
import unittest

class ValidAnagram(unittest.TestCase):
    def test(self):
        test_cases = [
            ("anagram", "nagaram", True),
            ("rat", "car", False),
            ("listen", "silent", True),
            ("hello", "world", False)
        ]
        for s, t, expected in test_cases:
            self.assertEqual(isAnagram(s, t), expected)
            self.assertEqual(isAnagram2(s, t), expected)
